pkg/antctl/raw/packetcapture/command.go

```
@@ -0,0 +1,334 @@
+}
+
+func packetCaptureRunE(cmd *cobra.Command, args []string) error {
+	option.timeout, _ = cmd.Flags().GetDuration("timeout")
+	if option.timeout > maxPacketCaptureTimeout {
+		return fmt.Errorf("timeout cannot be longer than %v", maxPacketCaptureTimeout)
+	}
+	if option.timeout == 0 {
+		option.timeout = defaultTimeout
+	}
+	if option.number == 0 {
+		return errors.New("packet number should be larger than 0")
+	}
+	out := cmd.OutOrStdout()
+
+	_, _, antreaClient, err := getClients(cmd)
+	if err != nil {
+		return err
+	}
+	pc, err := newPacketCapture()
+	if err != nil {
+		return fmt.Errorf("error when constructing a PacketCapture CR: %w", err)
+	}
+	createCtx, cancel := context.WithTimeout(cmd.Context(), 5*time.Second)
+	defer cancel()
+
+	if _, err := antreaClient.CrdV1alpha1().PacketCaptures().Create(createCtx, pc, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error when creating PacketCapture, is PacketCapture feature gate enabled? %w", err)
+	}
+
+	if option.nowait {
+		fmt.Fprintf(out, "PacketCapture Name: %s\n", pc.Name)
+		return nil
+	} else {
+		defer func() {
+			if err = antreaClient.CrdV1alpha1().PacketCaptures().Delete(context.TODO(), pc.Name, metav1.DeleteOptions{}); err != nil {
+				fmt.Fprintf(out, "error when deleting PacketCapture: %s", err.Error())
+			}
+		}()
+	}
+
+	var latestPC *v1alpha1.PacketCapture
+	var errMsg string
+	// add extra timeout to make sure the wait won't be interrupted before PacketCapture timeout.
+	err = wait.PollUntilContextTimeout(cmd.Context(), 1*time.Second, option.timeout+time.Second*5, false, func(ctx context.Context) (bool, error) {
+		res, err := antreaClient.CrdV1alpha1().PacketCaptures().Get(ctx, pc.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		for _, cond := range res.Status.Conditions {
+			if cond.Type == v1alpha1.PacketCaptureComplete && cond.Status == metav1.ConditionTrue {
+				latestPC = res
+				if cond.Reason == "Failed" {
+					errMsg = cond.Message
```

wouldn't it be better to just return an error here? Then you wouldn't need the `errors.New(errMsg)` below, which is a bit confusing IMO